### PR TITLE
Add plans for PostgreSQL by VSHN

### PR DESCRIPTION
**Pull request by glrf**
This PR adds plans for PostgreSQL. See user documentation here: https://github.com/appuio/appcat-docs/pull/44

We don't add `plus-X` plans to the component default as it is specific to APPUiO Cloud. I'll add it in the cluster catalog.

## Checklist

- [x] The PR has a meaningful title. It will be used to auto generate the
      changelog.
      The PR has a meaningful description that sums up the change. It will be
      linked in the changelog.
- [x] PR contains a single logical change (to build a better changelog).
- [x] Update the documentation. *Don't forget to generate the CRD API!*
- [x] Categorize the PR by adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.
- [x] Link this PR to related issues or PRs.

### CODE REVIEW

File: component/vshn_postgres.jsonnet [hunk 1]
```diff
@@ -38,7 +50,56 @@ local xrd = xrds.XRDFromCRD(
...
+                            ||| % [
+                              super.description,
+                              std.join(
+                                '\n\n  ',
+                                [
+                                  '%s - CPU: %s; Memory: %s; Disk: %s' % [ p, pgPlans[p].size.cpu, pgPlans[p].size.memory, pgPlans[p].size.disk ]
+                                  + if std.objectHas(pgPlans[p], 'note') && pgPlans[p].note != '' then ' - %s' % pgPlans[p].note else ''
+
+                                  for p in std.objectFields(pgPlans)
+                                ]
```
Comment (glrf) [inline component/vshn_postgres.jsonnet:85]: Not my pretties jsonnet, but it works and will result in proper validation and description in the CRD

File: component/vshn_postgres.jsonnet [hunk 2]
```diff
@@ -21,6 +21,18 @@ local certificateSecretName = 'tls-certificate';
 local serviceNameLabelKey = 'appcat.vshn.io/servicename';
 local serviceNamespaceLabelKey = 'appcat.vshn.io/claim-namespace';
 
+// Filter out disabled plans
+local pgPlans = std.foldl(
+  function(plans, key)
+    local p = pgParams.plans[key];
+    local enabled = p != null && p != {} && std.get(p, 'enabled', true);
```
Comment (zugao) [inline component/vshn_postgres.jsonnet:28]: I would add `enabled` flag by default in the defaults.yaml

File: component/vshn_postgres.jsonnet (hunk 1 above)
Comment (zugao) [reply component/vshn_postgres.jsonnet:85]: Better ugly that's working than otherwise :)

**Review by zugao — APPROVED** [review]
Some small things but overall it's good.

**Review by Kidswiss — APPROVED** [review]

**Pull request merged**